src-gui/src/main/java/org/openkilda/integration/converter/FlowConverter.java

```diff
@@ -62,14 +67,37 @@ public List<FlowInfo> toFlowsInfo(final List<Flow> flows) {
         }
         return null;
     }
-    
+
     /**
-    * To flowV2s info.
-    *
-    * @param flowV2s the flowV2s
-    * @return the list
-    */
-    public List<FlowInfo> toFlowV2sInfo(final List<FlowV2> flowV2s) { 
+     * To SwitchFlowsInfoPerPort info.
+     *
+     * @param switchFlowsPerPort switch flowsPer port
+     * @return the list
+     */
+    public SwitchFlowsInfoPerPort toFlowV2InfosPerPorts(final SwitchFlowsPerPort switchFlowsPerPort) {
+        if (switchFlowsPerPort == null || switchFlowsPerPort.getFlowsByPort() == null
+                || switchFlowsPerPort.getFlowsByPort().isEmpty()) {
+            return null;
+        }
+        final SwitchFlowsInfoPerPort switchFlowsInfoPerPort = new SwitchFlowsInfoPerPort();
+
+        Map<Integer, List<FlowInfo>> flowsByPortsMap = new HashMap<>();
+        switchFlowsPerPort.getFlowsByPort().forEach((port, flows) -> {
+            flowsByPortsMap.put(port, toFlowV2sInfo(flows));
+        });
```

Comment:
This can be done using java streams also.
```java
Map<Integer, List<FlowInfo>> flowsByPortsMap = switchFlowsPerPort.getFlowsByPort().entrySet().stream()
        .collect(Collectors.toMap(Map.Entry::getKey, e -> toFlowV2sInfo(e.getValue())));
```